xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -684,6 +712,32 @@ static void DisplayReconfigured(CGDirectDisplayID display,
 
       // associate with current window
       [appWindow setContentView:view];
+
+      // set the window to the appropriate screen and screen position
+      if (m_bFullScreen)
+      {
+        [appWindow setFrameOrigin:screen.frame.origin];
```

Comment:
I'm not certain, but I think the origin of a window that is presented in full screen cannot be changed, or has no effect.  Or worse, causes undefined behavior.

Check to see if there is macOS API to present a window on a specific screen.  I think there may be one.

Reply:
yeah in fullscreen if you try to change the origin point it doesn't work. In fact it's one of the things that is still left to do: if you change the monitor setting while in fullscreen kodi should move the window programmatically to the selected screen.
However it doesn't really affect the window creation since we always create it in windowed mode and then toggle fullscreen. Hence, here, kodi is still windowed; we move it to the proper screen and toggle fullscreen.